### PR TITLE
add ghex package name to git uri

### PR DESCRIPTION
**Pull request by halungge**
(FIX) Adding package name `ghex` to git uri for installation in requirements file.

This replaces [PR 583](https://github.com/C2SM/icon4py/pull/583)

Adding ghex package name to the git uri in the requirements file seems to fix the issue with the ghex build with  mpi4py 4.0.

(I don't really understand why it fixes the issue, but it is certainly not wrong to have to specify the package name`

*additional changes:*
fix tests to recent changes.

### CODE REVIEW

File: model/common/tests/grid_tests/mpi_tests/test_parallel_icon.py [hunk 1]
```diff
@@ -21,14 +22,19 @@
...
 try:
-    import mpi4py  # noqa F401:  import mpi4py to check for optional mpi dependency
+    import mpi4py  # F401:  import mpi4py to check for optional mpi dependency
 except ImportError:
     pytest.skip("Skipping parallel on single node installation", allow_module_level=True)
 
 
 @pytest.mark.parametrize("processor_props", [True], indirect=True)
 def test_props(processor_props):  # noqa: F811  # fixture
+    """dummy test to check whether the MPI initialization anc GHEX setup works."""
```
Comment (kotsaloscv) [inline model/common/tests/grid_tests/mpi_tests/test_parallel_icon.py:32]: and instead of anc ;)

**Review by kotsaloscv — APPROVED** [review]
LGTM

**Comment by halungge** [comment]
cscs-ci run dace

**Comment by halungge** [comment]
cscs-ci run default

**Comment by halungge** [comment]
cscs-ci run dace

**Comment by halungge** [comment]
cscs-ci run default

**Comment by halungge** [comment]
cscs-ci run dace

**Comment by halungge** [comment]
launch jenkins spack

**Comment by halungge** [comment]
launch jenkins spack spackBranch=py_gt4py_add_diskcache_dependency

**Comment by halungge** [comment]
launch jenkins spack spackProject=C2SM/py_gt4py_add_diskcache_dependency

**Comment by halungge** [comment]
launch jenkins spack spackProject=C2SM/py_gt4py_add_diskcache_dependency

**Comment by halungge** [comment]
launch jenkins spack spackBranch=py_gt4py_add_diskcache_dependency

**Comment by halungge** [comment]
cscs-ci run default

**Comment by halungge** [comment]
launch jenkins spack

**Pull request merged**